contracts/actions/insta/ClaimInstMaker.sol

```diff
@@ -0,0 +1,119 @@
+
+    IInstaMakerDAOMerkleDistributor public constant rewardDistributor =
+        IInstaMakerDAOMerkleDistributor(0xAC838332afc2937FdED89c16a59b2ED8e8e2743c);
+
+    address public constant INST_TOKEN_ADDR = 0x6f40d4A6237C257fff2dB00FA0510DeEECd303eb;
+    
+    struct Params {
+        uint256 index;
+        uint256 vaultId;
+        uint256 rewardAmount;
+        uint256 networthAmount;
+        bytes32[] merkleProof;
+        address to;
+    }
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory inputData = parseInputs(_callData);
+
+        _claimInst(inputData);
+
+        return bytes32(inputData.vaultId);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public payable override {
+        Params memory inputData = parseInputs(_callData);
+
+        _claimInst(inputData);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    function _claimInst(Params memory _inputData) internal {
+        address dsaAddress = instaAccountBuilder.build(address(this), 2, address(0));
+
+        mcdManager.give(_inputData.vaultId, dsaAddress);
+
+        rewardDistributor.claim(
+            _inputData.index,
+            _inputData.vaultId,
+            dsaAddress,
+            0x9cCf93089cb14F94BAeB8822F8CeFfd91Bd71649,
+            _inputData.rewardAmount,
+            _inputData.networthAmount,
+            _inputData.merkleProof
+        );
+
+        string[] memory _targetNames = new string[](2);
+        bytes[] memory _datas = new bytes[](2);
+        address _origin = address(this);
+
+        // connects dsaAccount with BASIC INST connector and transfers INST tokens
+        _targetNames[0] = "BASIC-A";
+        _datas[0] = abi.encodeWithSignature(
+            "withdraw(address,uint256,address,uint256,uint256)",
+            INST_TOKEN_ADDR,
+            type(uint).max,
+            _inputData.to,
+            0,
+            0
+        );
+        
+        // connects dsaAccount with MAKER INST connector and transfers ownership of the vault
+        _targetNames[1] = "MAKERDAO-A";
+        _datas[1] = abi.encodeWithSignature(
+            "transfer(uint256,address)",
+            _inputData.vaultId,
+            address(this)
+        );
+        
+        // calling fallback function of dsaAccount
+        (bool success, ) =
+            dsaAddress.call(abi.encodeWithSignature("cast(string[],bytes[],address)", _targetNames, _datas, _origin));
+        require(success, "fallback function call failed");
+
```

Comment:
Maybe a sanity check to see if the vault owner is address(this), as we are moving execution to a dsa spell